File: python/stack/online-stock-span.py
class StockSpanner:

    def __init__(self):
        self.prices = []

    def next(self, price: int) -> int:
        span = 1
        while self.prices:
            if self.prices[-1][0] > price:
                break
            curr = self.prices.pop()
            span += curr[1]

        self.prices.append((price, span))
        return span

# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)

File: c++/stack/online-stock-span.cc
class StockSpanner {
public:
    StockSpanner() {

    }

    int next(int price) {
        span_.push_back(price);

        int counter = 0;
        for (int i = span_.size() - 1; i >= 0; --i) {
            if (span_[i] <= price) {
                counter++;
            } else {
                break;
            }
        }

        return counter;
    }

private:
    std::vector<int> span_;
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */
